mewarpx/mewarpx/mwxrun.py

```diff
@@ -53,57 +53,86 @@ class MEWarpXRun(object):
+                - the minimum number of tiles via ``min_tiles``.
         """
-        lower_boundary_conditions = [field_boundary_xmin, field_boundary_zmin]
-        upper_boundary_conditions = [field_boundary_xmax, field_boundary_zmax]
-        lower_boundary_conditions_parts = [
-            parts_boundary_xmin, parts_boundary_zmin
-        ]
-        upper_boundary_conditions_parts = [
-            parts_boundary_xmax, parts_boundary_zmax
-        ]
-
-        self.grid = picmi.Cartesian2DGrid(
-            number_of_cells=[nx, nz],
-            lower_bound=[xmin, zmin],
-            upper_bound=[xmax, zmax],
-            lower_boundary_conditions=lower_boundary_conditions,
-            upper_boundary_conditions=upper_boundary_conditions,
-            lower_boundary_conditions_particles=lower_boundary_conditions_parts,
-            upper_boundary_conditions_particles=upper_boundary_conditions_parts,
-            warpx_max_grid_size=nz//min_tiles
+        # sanity check inputs
+        assert len(lower_bound) == self.dim
+        assert len(upper_bound) == self.dim
+        assert len(number_of_cells) == self.dim
+
+        boundary_conditions = {'fields': {}, 'particles': {}}
+
+        # set boundary conditions to be periodic in x and y except for RZ
+        if self.geom_str != 'RZ':
+            boundary_conditions['fields']['min'] = ['periodic'] * (self.dim - 1)
+            boundary_conditions['fields']['max'] = ['periodic'] * (self.dim - 1)
+            boundary_conditions['particles']['min'] = (
+                ['periodic'] * (self.dim - 1)
+            )
+            boundary_conditions['particles']['max'] = (
+                ['periodic'] * (self.dim - 1)
+            )
+        else:
+            boundary_conditions['fields']['min'] = ['none'] * (self.dim - 1)
+            boundary_conditions['fields']['max'] = ['none'] * (self.dim - 1)
+            boundary_conditions['particles']['min'] = (
+                ['absorbing'] * (self.dim - 1)
+            )
+            boundary_conditions['particles']['max'] = (
+                ['absorbing'] * (self.dim - 1)
+            )
+
+        # set default Z boundary conditions
+        boundary_conditions['fields']['min'].append('dirichlet')
+        boundary_conditions['fields']['max'].append('dirichlet')
+        boundary_conditions['particles']['min'].append('absorbing')
+        boundary_conditions['particles']['max'].append('absorbing')
+
+        # update boundary conditions if provided
+        for key in [key for key in kwargs.keys() if key.startswith('bc')]:
+            kind, dim, side = key.split('_')[-3:]
+            boundary_conditions[kind][side][self.coord_map[dim]] = kwargs[key]
+
+        if self.geom_str == 'RZ':
+            grid = picmi.CylindricalGrid
+        elif self.dim == 1:
+            grid = picmi.Cartesian1DGrid
+        elif self.dim == 2:
+            grid = picmi.Cartesian2DGrid
+        elif self.dim == 3:
+            grid = picmi.Cartesian3DGrid
```

Comment:
I would check geom_str explicitly for Z/XZ/XYZ, then error if it's different, in case we later add a new geometry.